File: backend/schemas/rating.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field


class RatingCreate(BaseModel):
    product_id: UUID
    rating: int = Field(ge=1, le=5)
    comment: str | None = Field(default=None, max_length=1000)


class RatingUpdate(BaseModel):
    rating: int = Field(ge=1, le=5)
    comment: str | None = Field(default=None, max_length=1000)


class RatingResponse(BaseModel):
    id: UUID
    product_id: UUID
    rating: int
    comment: str | None
    photo_url: str | None = None
    user_name: str
    user_pic: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/types/rating.ts
export interface Rating {
  id: string;
  product_id: string;
  rating: number;
  comment: string | null;
  photo_url: string | null;
  user_name: string;
  user_pic: string | null;
  created_at: string;
  updated_at: string;
}

export interface RatingInput {
  product_id: string;
  rating: number;
  comment: string;
  photo?: File | null;
}